sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/Transport.java

```diff
@@ -101,7 +102,7 @@ public static Storage.Builder newStorageClient(GcsOptions options) {
                     // logging it by default clutters the output during file staging.
                     new RetryHttpRequestInitializer(
                         ImmutableList.of(404), new UploadIdResponseInterceptor())))
-            .setApplicationName(options.getAppName())
+            .setApplicationName(applicationNamePrefix + options.getAppName())
```

Comment:
This would actually make an invalid user agent if it came first. Is there a spec for where the application name goes in the user-agent field?

Reply:
Fixed.